File: js/src/Blockchain/Blockchain.Abstraction/Interfaces/ISwapWorkflow.ts
import { defineSignal, defineUpdate } from '@temporalio/workflow';
import { HTLCCommitEventMessage, HTLCLockEventMessage } from '../Models/EventModels/HTLCBlockEventResposne';
import { AddLockSignatureRequest } from '../Models/TransactionBuilderModels/AddLockSignatureRequest';

export const lockCommitedSignal = defineSignal<[HTLCLockEventMessage]>('LockCommited');

export const setAddLockSigUpdate = defineUpdate<boolean, [AddLockSignatureRequest]>('SetAddLockSig');

export interface ISwapWorkflow {
  LockCommited: typeof lockCommitedSignal;
  SetAddLockSig: typeof setAddLockSigUpdate;
  RunAsync(message: HTLCCommitEventMessage): Promise<void>;
}

File: csharp/src/Workflows.Abstractions/Workflows/ISwapWorkflow.cs
using Temporalio.Workflows;
using Train.Solver.Workflows.Abstractions.Models;

namespace Train.Solver.Workflows.Abstractions.Workflows;

[Workflow]
public interface ISwapWorkflow
{
    [WorkflowSignal]
    Task LockCommitedAsync(HTLCLockEventMessage message);

    [WorkflowRun]
    Task RunAsync(HTLCCommitEventMessage message);

    [WorkflowUpdate]
    Task<bool> SetAddLockSigAsync(AddLockSignatureRequest addLockSig);
}